src/Application/ServerApplication.php

```diff
@@ -107,10 +107,15 @@ public function addClient(Protocol\LivereloadProtocol $client)
 
     public function removeClient(Protocol\LivereloadProtocol $client)
     {
+        $this->clients = array_values(array_filter($this->clients, function($client) {
+            return !empty($client->app) && !empty($client->conn);
+        }));
+        
         $index = array_search($client, $this->clients, true);
         if($index == false){
```

Comment:
You are right, that seems to be the cause all along. I've seen you have updated the master branch with this change, and everything seems to just run smoothly. Thanks!